packages/share_plus/share_plus/lib/share_plus.dart

```diff
@@ -87,12 +87,17 @@ class Share {
   /// any other call to any share method that returns a result _might_ result in
   /// a [PlatformException] (on Android).
   ///
-  /// Because IOS, Android and macOS provide different feedback on share-sheet interaction,
-  /// a result on IOS will be more specific than on Android or macOS. While IOS can detect if
-  /// the user actually completed his selected action or aborted it midway, Android and macOS
-  /// only record if the user selected an action or outright dismissed the share-sheet.
+  /// Because IOS, Android and macOS provide different feedback on share-sheet
+  /// interaction, a result on IOS will be more specific than on Android or macOS.
+  /// While on IOS the selected action can inform its caller that it was completed
+  /// or dismissed midway (_actions are free to return whatever they want_),
+  /// Android and macOS only record if the user selected an action or outright
+  /// dismissed the share-sheet.
```

Comment:
I think we should clearly state the fact that on Android and macOS we can't guarantee share action as it is something that definitely requested by people using `withResult` calls. 

```suggestion
  /// Android and macOS only record if the user selected an action or outright
  /// dismissed the share-sheet. It is not guaranteed that user actually shared something.
```